Tasks/KubernetesManifestV0/src/actions/promote.ts

```diff
@@ -13,9 +13,20 @@ export async function promote(ignoreSslErrors?: boolean) {
     const kubectl = new Kubectl(await utils.getKubectl(), TaskInputParameters.namespace, ignoreSslErrors);
 
     if (canaryDeploymentHelper.isCanaryDeploymentStrategy()) {
+        if (canaryDeploymentHelper.isTrafficSplitCanaryStrategy()) {
+            tl.debug('Redirecting traffic to canary deployment.');
```

Comment:
Why are we redirecting traffic to Canary deployment for 'promote' action ?

Note -  Canary pods count will be less. Redirecting entire traffic to canary how are we making sure it will be able to handle the load?

Reply:
This is up to customer.  He needs to set canary deployment with required load and then do promote.